### PR TITLE
feat(TDI-45915): support Analytics queries for Couchbase

### PR DESCRIPTION
**Pull request by dmytrogrygorenko**
https://jira.talendforge.org/browse/TDI-45915
Implement support of Analytics queries for existing Couchbase components (TCK).

### CODE REVIEW

File: couchbase/src/main/java/org/talend/components/couchbase/source/CouchbaseInput.java [hunk 1]
```diff
@@ -90,79 +98,99 @@ public void init() {
...
-                statement = asPath.limit(Integer.parseInt(configuration.getLimit().trim()));
+        if (configuration.getSelectAction() == SelectAction.ANALYTICS) {
+            AnalyticsQuery analyticsQuery = AnalyticsQuery.simple(configuration.getQuery());
+            AnalyticsQueryResult queryResult = bucket.query(analyticsQuery);
+            if (queryResult.finalSuccess()) {
+                analytics_index = queryResult.rows();
             } else {
-                statement = asPath;
+                LOG.error(i18n.queryResultError());
+                throw new IllegalArgumentException(queryResult.errors().toString());
```
Comment (mbasiuk-talend) [inline couchbase/src/main/java/org/talend/components/couchbase/source/CouchbaseInput.java:108]: As far as I know, we have to throw ComponentException

File: couchbase/src/main/java/org/talend/components/couchbase/source/CouchbaseInput.java [hunk 2]
```diff
@@ -90,79 +98,99 @@ public void init() {
...
-            if (!configuration.getLimit().isEmpty()) {
-                statement = asPath.limit(Integer.parseInt(configuration.getLimit().trim()));
+        if (configuration.getSelectAction() == SelectAction.ANALYTICS) {
+            AnalyticsQuery analyticsQuery = AnalyticsQuery.simple(configuration.getQuery());
+            AnalyticsQueryResult queryResult = bucket.query(analyticsQuery);
+            if (queryResult.finalSuccess()) {
+                analytics_index = queryResult.rows();
             } else {
-                statement = asPath;
+                LOG.error(i18n.queryResultError());
```
Comment (mbasiuk-talend) [inline couchbase/src/main/java/org/talend/components/couchbase/source/CouchbaseInput.java:107]: You could also pass as a parameter the queryResult errors

File: couchbase/src/main/java/org/talend/components/couchbase/source/CouchbaseInput.java [hunk 3]
```diff
@@ -90,79 +98,99 @@ public void init() {
...
-                    return documentParser.parse(bucket, id);
-                } catch (TranscodingException e) {
-                    // document is not a non-json, try to get next document
                 }
             }
         }
         return null;
     }
 
-    private void checkErrors(N1qlQueryResult n1qlQueryRows) {
```
Comment (mbasiuk-talend) [inline couchbase/src/main/java/org/talend/components/couchbase/source/CouchbaseInput.java:155]: You could keep this method and try to adopt for current impl.

**Review by mbasiuk-talend — COMMENTED** [review]
minor comments

File: couchbase/src/main/java/org/talend/components/couchbase/source/CouchbaseInput.java (hunk 1 above)
Comment (dmytrogrygorenko) [reply couchbase/src/main/java/org/talend/components/couchbase/source/CouchbaseInput.java]: Done. Fixed in every occurrences.

File: couchbase/src/main/java/org/talend/components/couchbase/source/CouchbaseInput.java (hunk 2 above)
Comment (dmytrogrygorenko) [reply couchbase/src/main/java/org/talend/components/couchbase/source/CouchbaseInput.java]: Done. Method signature was changed to log actual error messages.

File: couchbase/src/main/java/org/talend/components/couchbase/source/CouchbaseInput.java (hunk 3 above)
Comment (dmytrogrygorenko) [reply couchbase/src/main/java/org/talend/components/couchbase/source/CouchbaseInput.java:155]: Fixed. Refactored to separate method.

**Review by mbasiuk-talend — DISMISSED** [review]
LGTM

**Review by mbasiuk-talend — APPROVED** [review]
LGTM

**Pull request merged**